dist/tools/pkg_version_check/version_check.py

```diff
@@ -66,50 +59,41 @@ def get_makefiles():
 
 
 def process_url(repo_url):
-    """
-    Remove the '.git' suffix from the repository URL if it is present.
-    """
+    # Remove the '.git' suffix from the repository URL if it is present
     if repo_url.endswith(".git"):
         repo_url = repo_url[:-4]
     return repo_url
 
 
-def get_hash_from_branch(repo_url, branch):
-    """
-    Get the latest commit hash from the specified branch of the GitHub repository.
-    """
+def get_hash_from_branch(repo_url, branch, headers):
+    # Get the latest commit hash from the specified branch of the GitHub repository
     commit_url = f"{repo_url}/commits/{branch}"
     commit_response = requests.get(commit_url, headers=headers)
     commit_response.raise_for_status()  # Raise an exception for HTTP errors
     commit_data = commit_response.json()
     return commit_data["sha"]
 
 
-def get_latest_hash(repo_url):
-    """
-    Get the latest commit hash from the GitHub repository, prioritizing the main branch first and the master branch if necessary.
-    """
+def get_latest_hash(repo_url, headers):
+    # Get the latest commit hash from the GitHub repository
     repo_url = process_url(repo_url)
     # Try to get the commit hash from the main branch
     try:
-        return get_hash_from_branch(repo_url, "main")
+        return get_hash_from_branch(repo_url, "main", headers)
     except requests.HTTPError:
         # If the main branch is not found, try to get the commit hash from the master branch
         try:
-            return get_hash_from_branch(repo_url, "master")
+            return get_hash_from_branch(repo_url, "master", headers)
         except requests.HTTPError:
             try:
-                return get_hash_from_branch(repo_url, "dev")
+                return get_hash_from_branch(repo_url, "dev", headers)
             except Exception:
                 print("Failed to fetch the latest commit hash")
                 return None
 
 
-def get_release_or_commit_hash(repo_url):
-    """
-    Get the commit hash of the latest release from the GitHub repository.
-    If there is no release, get the latest commit hash from the main branch, and then master branch if necessary.
-    """
+def get_release_or_commit_hash(repo_url, headers):
+    # Get the commit hash
```

Comment:
You could have left the function documentation in a docstring, I don't think that'd be against the coding conventions.